### DISCUSSION TITLE
EthereumProvider.init want switch the chain

### DISCUSSION THREAD

**Discussion by windatang** [WalletConnect/walletconnect-monorepo#2725] [Developer Support]
Dev for web-app & debug with mobile-app  I report that 

Register multiple Chain will reject by app
[issues/2491](https://github.com/WalletConnect/walletconnect-monorepo/issues/2491)
Register one Chain only create instance is working otherwise it no way to switch (@walletconnect/modal) 
[issues/2661](https://github.com/WalletConnect/walletconnect-monorepo/issues/2661)
- try the way reset and setChainIds 
- try the way  use the `"@walletconnect/modal"` -> `@web3modal/core` ->  `ConfigCtrl.setConfig`
Both are not working

**Comment by glitch-txs** [comment]
Hi, could you provide a minimal reproducible example for both cases?

**Reply by windatang** [reply]
https://codesandbox.io/p/sandbox/walletconnect2-0-p28lvm

**Reply by windatang** [reply]
try to directory change rpc.chains as `ethereumProvider.rpc.chains = [ethereumProvider.formatChainId(Number(props.chainId ?? 1))];`
https://github.com/WalletConnect/walletconnect-monorepo/issues/2661#issuecomment-1603995228
